<?xml version="1.0" ?>

<!-- 
MFC: based on Adam's spawn actor from RASberry
-->

<launch>  
  <arg name="use_gpu_gazebo_lidar" default="false"/>
  <arg name="actor_name" default="actor"/>
  <arg name="map_topic" default="/map"/>
  <arg name="global_frame_id" default="world"/>
  <arg name="start_pose_x" default="0"/>
  <arg name="start_pose_y" default="0"/>
  <arg name="start_pose_z" default="0"/>
  <arg name="start_pose_R" default="0"/>
  <arg name="start_pose_P" default="0"/>
  <arg name="start_pose_Y" default="0"/>

  <arg name="actor_trajectory_file" default="$(find moving_actor_gazebo)/config/trajectories/trajectory.json"/>

  <arg name="actor_xacro_file" default="$(find moving_actor_gazebo)/urdf/actor.xacro"/>
  
  <group ns="$(arg actor_name)">

    <param name="robot_description" command="$(find xacro)/xacro '$(arg actor_xacro_file)' '--inorder' actor_name:='$(arg actor_name)' tf_prefix:='$(arg actor_name)' use_gpu:='$(arg use_gpu_gazebo_lidar)'"/>

    <node pkg="gazebo_ros" type="spawn_model" name="$(arg actor_name)_spawner" output="screen" args="-x $(arg start_pose_x) -y $(arg start_pose_y) -z $(arg start_pose_z) -R $(arg start_pose_R)  -P 		      $(arg start_pose_P)  -Y $(arg start_pose_Y) -urdf -param /$(arg actor_name)/robot_description -model $(arg actor_name)"/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>


    <node pkg="tf" type="static_transform_publisher" name="world_2_actor_odom_tfb"
            args="0.0 0.0 0.0  0.0 0.0 0.0 $(arg global_frame_id) $(arg actor_name)/odom 20"/>
      
    <include file="$(find moving_actor_gazebo)/launch/actor_move_base_teb.launch">
      <arg name="actor_name" value="$(arg actor_name)"/>
      <arg name="map_topic" value="$(arg map_topic)"/>
    </include>

    <node pkg="moving_actor_gazebo" type="goal_feeder.py" name="goal_manager">
      <param name="goals_filename" type="string" value="$(arg actor_trajectory_file)"/>
      <param name="frame_id" type="string" value="$(arg global_frame_id)"/>
      <param name="robot_base_frame_id" type="string" value="$(arg actor_name)/base_link"/>
      <param name="move_base_actionserver_name" type="string" value="/$(arg actor_name)/move_base"/>
      <param name="wait_time" type="double" value="0.00005"/>
      <param name="interpolate" type="boolean" value="False"/>
      <param name="loop" type="boolean" value="True"/>
      <param name="visual_pub_topic_name" type="string" value="/$(arg actor_name)/markers"/>
    </node>
   </group>
</launch> 
